libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,25 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
+        if (!baro->all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (checks_to_perform & ARMING_CHECK_GPS) {
+            const AP_GPS *gps = AP_GPS::get_singleton();
+            if (gps->status() >= AP_GPS::GPS_OK_FIX_3D && gps->get_vdop() < 170 && gps->num_sats() > 8) {
+                float baro_altasl = baro->get_altitude_difference(101325.0f, baro->get_pressure());
+                float gps_alt = gps->location().alt*0.01f;
+                if (baro_altasl > (gps_alt+100.0f) || baro_altasl < (gps_alt-100.0f)) {
```

Comment:
Yeah, this is way to tight. Start by changing the math to just be `fabsf(baro_alt - gps_alt) > THRESHOLD`. The threshold you need to be looking at for this needs to be somewhere north of 1200 meters (probably more like 1,500 meters). As is this blocks high density altitude days incorrectly.

Reply:
1500m is really too much, the issue I previously had was few hundred meter difference which caused the crash.
I am thinking about making it relative to alt. Like 200m error for each 1000m above seal level. What do you think?
I have already done hours of flying with this patch. But I've found that some boards with ms5607 cannot pass this check.
It's an old issue with not resolution, but I'm sending a proposed one.

Reply:
Check now. I've set it some how relative to alt.